lib/ansible/modules/remote_management/redfish/redfish_facts.py

```diff
@@ -0,0 +1,217 @@
+        elif command == "GetPsuInventory":
+            result = rf_utils.get_psu_inventory()
+        elif command == "GetCpuInventory":
+            result = rf_utils.get_cpu_inventory()
+        elif command == "GetNicInventory":
+            result = rf_utils.get_nic_inventory()
+
+        # Storage
+        elif command == "GetStorageControllerInventory":
+            result = rf_utils.get_storage_controller_inventory()
+        elif command == "GetDiskInventory":
+            result = rf_utils.get_disk_inventory()
+
+        # Chassis
+        elif command == "GetFanInventory":
+            # execute only if we find Chassis resource
+            result = rf_utils._find_chassis_resource(rf_uri)
+            if result['ret'] is False:
+                module.fail_json(msg=result['msg'])
+            result = rf_utils.get_fan_inventory()
+
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Accounts":
+        # execute only if we find an Account service resource
+        result = rf_utils._find_accountservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        if command == "ListUsers":
+            result = rf_utils.list_users()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "System":
+        # execute only if we find a System resource
+        result = rf_utils._find_systems_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        if command == "GetBiosAttributes":
+            result = rf_utils.get_bios_attributes()
+        elif command == "GetBiosBootOrder":
+            result = rf_utils.get_bios_boot_order()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Update":
+        # execute only if we find UpdateService resources
+        result = rf_utils._find_updateservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        if command == "GetFirmwareInventory":
+            result = rf_utils.get_firmware_inventory()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    elif category == "Manager":
+        # execute only if we find a Manager service resource
+        result = rf_utils._find_managers_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        if command == "GetManagerAttributes":
+            result = rf_utils.get_manager_attributes()
+        elif command == "GetLogs":
+            result = rf_utils.get_logs()
+        else:
+            result = {'ret': False, 'msg': 'Invalid Command'}
+
+    else:
+        result = {'ret': False, 'msg': 'Invalid Category'}
+
+    # Return data back or fail with proper message
+    if result['ret'] is True:
+        del result['ret']
+        module.exit_json(result=result)
```

Comment:
Facts modules need to return their results inside the `ansible_facts` key in order to be set as facts. I'd suggest something line `redfish:{"inventory":[], "accounts":[], "system":[], "updates":[], "chassis":[]}`, or `redfish_inventory:[]`, `redfish_accounts:[]`, etc. I'm not sure if the types of those facts are lists of dicts, but I hope you get the idea. Feel free to ask for clarification if I didn't explain that very well.

Reply:
Information returned is a dict (i.e. firmware inventory) or a list of dicts if more than one item (information for fans). Updating to ansible_facts works fine for me. Example:
```
{
    "ansible_facts": {
        "entries": [
            {
                "Health": "OK", 
                "Name": "System Board Fan1", 
                "RPMs": 5040, 
                "State": "Enabled"
            }, 
            {
                "Health": "OK", 
                "Name": "System Board Fan2", 
                "RPMs": 5040, 
                "State": "Enabled"
            }, 
            {
                "Health": "OK", 
                "Name": "System Board Fan3", 
                "RPMs": 5160, 
                "State": "Enabled"
            }, 
            {
                "Health": "OK", 
                "Name": "System Board Fan4", 
                "RPMs": 5040, 
                "State": "Enabled"
            }, 
            {
                "Health": "OK", 
                "Name": "System Board Fan5", 
                "RPMs": 5040, 
                "State": "Enabled"
            }, 
            {
                "Health": "OK", 
                "Name": "System Board Fan6", 
                "RPMs": 5040, 
                "State": "Enabled"
            }
        ]
    }, 
    "changed": false, 
    "failed": false
}
```